packages/taquito-signer/src/derivation-tools/ecdsa.ts

```diff
@@ -0,0 +1,136 @@
+/* eslint-disable @typescript-eslint/no-this-alias */
+import { ec, curve } from "elliptic";
+import { Hard, ExtendedPrivateKey } from "./index";
+import { HMAC } from "@stablelib/hmac";
+import { SHA512 } from "@stablelib/sha512";
+import BN from "bn.js";
+import { parseHex } from "./utils";
+import { InvalidBitSize, InvalidCurveError, InvalidSeedLengthError, PrivateKeyError} from "../errors";
+
+export type CurveName = "p256" | "secp256k1";
+
+const seedKey: Record<CurveName, string> = {
+    p256: "Nist256p1 seed",
+    secp256k1: "Bitcoin seed",
+};
+
+interface KeyPair extends ec.KeyPair {
+    priv: BN | null;
+    pub: curve.base.BasePoint | null;
+}
+
+// MinSeedSize is the minimal allowed seed byte length
+const minSeedSize = 16;
+// MaxSeedSize is the maximal allowed seed byte length
+const maxSeedSize = 64;
+
+export class PrivateKey implements ExtendedPrivateKey {
+    readonly keyPair: KeyPair;
+/**
+ *
+ * @param priv key pair priv (BN) pub (curve.base.BasePint) if applicable
+ * @param chainCode slice 32->n HMAC hash key and seedkey (first instance curve default seedKey. after hmac value slice 32->n)
+ */
+    constructor(priv: ec.KeyPair, public readonly chainCode: Uint8Array) {
+        this.keyPair = <KeyPair>priv;
+    }
+    /**
+     * @param seedSrc result of Bip39.mnemonicToSeed
+     * @param curve known supported curve p256 or secp256k1
+     * @returns instance of PrivateKey non-HD keys derived
+     */
+    static fromSeed(seedSrc: Uint8Array | string, curve: CurveName): PrivateKey {
+        let seed = typeof seedSrc === "string" ? parseHex(seedSrc) : seedSrc;
+        if (seed.length < minSeedSize || seed.length > maxSeedSize) {
+            throw new InvalidSeedLengthError(seed.length);
+        }
+        if (!Object.prototype.hasOwnProperty.call(seedKey, curve)) {
+            throw new InvalidCurveError(`unknown curve ${curve}`);
+        }
+        const c = new ec(curve);
+        if (c.n?.bitLength() !== 256) {
+            throw new InvalidBitSize(`invalid curve bit size ${c.n?.bitLength()}`);
+        }
+
+        const key = new TextEncoder().encode(seedKey[curve]);
+        let d: BN | null = null;
+        let chain: Uint8Array = new Uint8Array();
+        let i = 0;
+        while (i === 0) {
+            const sum = new HMAC(SHA512, key).update(seed).digest();
```

Comment:
I think the indentation in this file are 4 spaces instead of 2. We should stick to 2.

Reply:
I think running the linter through this file would fix the semicolons and the indentation differences

Reply:
neglected to format the doc after copying it over 😩